app/code/core/Mage/Core/Model/Config.php

```diff
@@ -474,20 +475,9 @@ protected function _canUseLocalModules()
      */
     protected function _canUseCacheForInit()
     {
-        if (Mage::app()->useCache('config') && $this->_allowCacheForInit) {
-            $retries = 10;
-            do {
-                if ($this->_loadCache($this->_getCacheLockId())) {
-                    if ($retries) {
-                        usleep(500000); // 0.5 seconds
-                    }
-                } else {
-                    return true;
-                }
-            } while ($retries--);
-        }
-
-        return false;
+        return $this->_allowCacheForInit
+            && Mage::isInstalled(['etc_dir' => $this->getOptions()->getEtcDir()])
+            && Mage::app()->useCache('config');
```

Comment:
Is it important to preserve the check sequence?

```suggestion
        return Mage::isInstalled(['etc_dir' => $this->getOptions()->getEtcDir()])
            && Mage::app()->useCache('config')
            && $this->_allowCacheForInit;
```

Reply:
It's a micro-optimization.. I check the class property first since that is the fastest so it short-circuits the remaining slower methods. But the vast majority of cases they will all be true so I suppose it won't make any difference, just a habit.. I maintained the original order between isInstalled and useCache in case there is some reason to, but I can't imagine checking a declared class property first can have any negative effects in this case.